src/security/builtin_plugins/cryptographic/src/crypto_transform.c

```diff
@@ -1191,9 +1229,17 @@ static DDS_Security_boolean encode_rtps_message_encrypt (
   size_t size;
   uint32_t transform_kind, transform_id;
 
+  DDSRT_STATIC_ASSERT(num_segs == 3);
```

Comment:
Is this really accepted by all compilers? It could be, but I am not sure and the CI wasn't run for this commit (I suppose because it needs to be rebased first)